### ISSUE TITLE
TypeError: import_react.default.act is not a function

### ISSUE THREAD

**Issue by lgaspari** [vitest-dev/vitest#6771]
### Describe the bug

It seems like I'm not able to run Vitest in my local due to a weird `TypeError`. I've never seen this error nor I can find anything online.

```
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 1 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  |0| vitest-example/HelloWorld.test.tsx > renders name
TypeError: import_react.default.act is not a function
 ❯ vitest-example/HelloWorld.test.tsx:6:24
      4| 
      5| test('renders name', async () => {
      6|   const { getByText } = render(<HelloWorld name="Vitest" />);
       |                        ^
      7|   await expect.element(getByText('Hello Vitest!')).toBeInTheDocument();
      8| });

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯

 Test Files  1 failed (1)
      Tests  1 failed (1)
   Start at  09:25:16
   Duration  1.40s (transform 0ms, setup 0ms, collect 53ms, tests 2ms, environment 0ms, prepare 13ms)
```

### Reproduction

1. Install Vitest `npm install -D vitest`
2. Install Vitest Browser `npx vitest init browser`
3. Run `npm run test:browser`

### System Info

```shell
System:
    OS: macOS 14.5
    CPU: (8) arm64 Apple M1
    Memory: 86.86 MB / 8.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.13.1 - ~/.nvm/versions/node/v20.13.1/bin/node
    Yarn: 1.22.22 - ~/.yarn/bin/yarn
    npm: 10.5.2 - ~/.nvm/versions/node/v20.13.1/bin/npm
    pnpm: 8.14.1 - ~/.nvm/versions/node/v20.13.1/bin/pnpm
    bun: 1.1.3 - ~/.bun/bin/bun
    Watchman: 2024.04.15.00 - /opt/homebrew/bin/watchman
  Browsers:
    Brave Browser: 129.1.70.123
    Chrome: 129.0.6668.101
    Edge: 117.0.2045.40
    Safari: 17.5
  npmPackages:
    @vitejs/plugin-react: ^4.2.1 => 4.2.1 
    @vitest/browser: ^2.1.3 => 2.1.3 
    vite: ^5.1.6 => 5.1.6 
    vitest: ^2.1.3 => 2.1.3
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
What react version do you use?

**Comment by lgaspari** [comment]
> What react version do you use?

I'm currently using `"react": "^18.2.0"` and `"react-dom": "^18.2.0"`

**Comment by sheremet-va** [comment]
This is fixed in `vitest-browser-react` 0.0.3. Turns out `react` documented to use `React.act`, but didn't export it: https://github.com/facebook/react/pull/28916

**Comment by lgaspari** [comment]
> This is fixed in `vitest-browser-react` 0.0.3. Turns out `react` documented to use `React.act`, but didn't export it: [facebook/react#28916](https://github.com/facebook/react/pull/28916)

OMG I was going nuts with this issue that came out of nowhere. Upgrading `vitest-browser-react` from `0.0.2` to `0.0.3`, thanks a lot @sheremet-va ❤️.

```
 RERUN  vitest-example/HelloWorld.test.tsx x1

 ✓ |0| vitest-example/HelloWorld.test.tsx (1)
   ✓ renders name

 Test Files  1 passed (1)
      Tests  1 passed (1)
   Start at  10:07:53
   Duration  10ms
```

Just to understand a bit deeper this issue. Is it because I'm using React `18.2.0` instead of `18.3.0` or is it because `vitest-browser-react` had to consider the case where `act` is not properly exported?

**Comment by sheremet-va** [comment]
> Is it because I'm using React `18.2.0` instead of `18.3.0` or is it because `vitest-browser-react` had to consider the case where `act` is not properly exported?

`vitest-browser-react` is supposed to work with React > 18.0, so the bug is there, not in your code.
